mesonbuild/coredata.py

```diff
@@ -117,6 +117,22 @@ def toint(self, valuestring):
     def validate_value(self, value):
         return self.toint(value)
 
+class UserUmaskOption(UserIntegerOption):
+    def __init__(self, name, description, value, yielding=None):
+        super().__init__(name, description, 0, 0o777, value, yielding)
+
+    def set_value(self, newvalue):
+        if newvalue is None or newvalue == 'none':
+            self.value = None
+        else:
+            super().set_value(newvalue)
+
+    def toint(self, valuestring):
+        try:
+            return int(valuestring, 8)
+        except ValueError:
+            raise MesonException('Value string "%s" is not convertable to an integer.' % valuestring)
```

Comment:
Typo: convertable. But I think it's better to just reuse the error message that python gives you, it's more precise:
```python
except ValueError as e:
  raise MesonException(str(e))
```

Reply:
`MesonException('Invalid mode: {}'.format(e))` would be even better.

Reply:
Done.